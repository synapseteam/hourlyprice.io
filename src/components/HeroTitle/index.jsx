import React from "react";
import { useCustomTranslation } from "i18n";

export default function HeroTitle() {
  const [t, Trans] = useCustomTranslation();

  return (
    <>
      <h1>{t("heroTitle")}</h1>
      <h3>
        <Trans components={{ citation: <q /> }}>heroSubTitle</Trans>
      </h3>
    </>
  );
}
