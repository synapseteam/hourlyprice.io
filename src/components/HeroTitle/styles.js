import { css } from "@emotion/react";

export const styles = {
  container: css`
    width: 100%;
  `,
  title: css`
    font-style: italic;
    line-height: 1;
    margin: 1.4rem 0 0 0;
    @media (max-width: 720px) {
      margin: 0;
    }
  `,
  subTitle: css`
    font-size: 1rem;
    font-style: italic;
    font-weight: 400;
    margin-bottom: 1.8rem;
  `,
};
