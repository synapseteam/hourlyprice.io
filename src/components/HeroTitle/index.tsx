/** @jsxImportSource @emotion/react */
import { useTranslation, Trans } from "react-i18next";
import { styles } from "./styles";

const HeroTitle: React.FC = (): JSX.Element => {
  const [t] = useTranslation();

  return (
    <div css={styles.container}>
      <h1 css={styles.title}>{t("heroTitle")}</h1>
      <h3 css={styles.subTitle}>
        <Trans components={{ citation: <q /> }}>heroSubTitle</Trans>
      </h3>
    </div>
  );
};
export default HeroTitle;
