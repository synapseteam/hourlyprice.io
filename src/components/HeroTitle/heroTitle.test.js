import React from "react";
import renderer from "react-test-renderer";

import HeroTitle from "components/HeroTitle";

if (!global.React) {
  global.React = React;
}

it("Hero title rendered correctly", () => {
  const tree = renderer.create(<HeroTitle />).toJSON();

  expect(tree).toMatchSnapshot();
});
