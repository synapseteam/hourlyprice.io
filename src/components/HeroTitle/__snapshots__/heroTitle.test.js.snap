// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero title rendered correctly 1`] = `
Array [
  <h1>
    Hourly Rate Calculator
  </h1>,
  <h3>
    <q>
      Time is money
    </q>
     as Benjamin Franklin Said.
  </h3>,
]
`;
